tally_ho/apps/tally/models/result_form.py

```diff
@@ -422,6 +423,61 @@ def reject(
         self.reject_reason = reject_reason
         self.save()
 
+    def reset_to_unsubmitted(self, user=None, reason=None):
+        """Reset form to UNSUBMITTED state and deactivate all related records.
+
+        This method deactivates all results, reconciliation forms, audits,
+        clearances, and quality control records associated with this form,
+        then sets the form state to UNSUBMITTED.
+
+        Args:
+            user: UserProfile object of the user performing the reset
+            reason: Text reason for resetting the form
+        """
+
+        # Deactivate all results
+        for result in self.results.all():
+            result.active = False
+            result.save()
```

Comment:
Would using the [queryset update()](https://docs.djangoproject.com/en/5.2/ref/models/querysets/#django.db.models.query.QuerySet.update) be more efficient here? Can we also only update active records? Note, we may already have some inactive records that we should not touch during the reset.

```python
self.results.filter(active=True).update(active=False)
```
It may also be important to link all these changes using the `modified_date` field.
```Python
..
modified_date = timezone.now()
self.results.filter(active=True).update(active=False, modified=modified_date)
...
```